domain/controllernode/state/state.go

```diff
@@ -146,3 +148,90 @@ WHERE  namespace = $dbNamespace.namespace`, dbNamespace)
 
 	return namespace, nil
 }
+
+// SetRunningAgentBinaryVersion sets the running agent binary version for the
+// provided controllerID. Any previously set values for this controllerID will
+// be overwritten by this call.
+//
+// The following errors can be expected:
+// - [controllernodeerrors.NotFound] if the controller node does not exist.
+// - [coreerrors.NotSupported] if the architecture is not found in the architecture database table.
```

Comment:
Not sure we need to be prescriptive in how the implementation works. we just need to say if the architecture is unknown  etc.

Reply:
I see, sure will change this.